src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,72 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+
+            session.persist(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert movieSession " + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return Optional.ofNullable(session.get(MovieSession.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get a movie session by id:" + id, e);
+        }
+    }
+
+    @Override
+    public List<MovieSession> getAllAvailableSessions(Long id, LocalDate date) {
+        LocalDateTime startOfDay = date.atStartOfDay();
+        LocalDateTime endOfDay = date.atTime(LocalTime.MAX);
+        String hql = "SELECT ms FROM MovieSession ms "
+                + "JOIN FETCH ms.movie "
+                + "WHERE ms.movie.id = :movieId "
+                + "AND ms.showTime BETWEEN :startOfDay AND :endOfDay";
+
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+
+            Query<MovieSession> query = session.createQuery(
+                    hql, MovieSession.class);
+            query.setParameter("movieId", id);
+            query.setParameter("startOfDay", startOfDay);
+            query.setParameter("endOfDay", endOfDay);
+            return query.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get a movie sessions", e);
```

Comment:
The error message in the `DataProcessingException` for the `getAllAvailableSessions` method could be more descriptive. Consider including more context about the operation that failed, such as the method name or operation type.